spec/filters/multiline_spec.rb

```diff
@@ -150,4 +161,120 @@
       insist { subject[1]["message"] } == "  match2\nnomatch2"
     end
   end
+
+  describe "keep duplicates by default on message field" do
+    config <<-CONFIG
+      filter {
+        multiline {
+          pattern => "^\s"
+          what => "next"
+          add_tag => ["multi"]
+        }
+      }
+    CONFIG
+
+    sample ["  match1", "  match1", "nomatch1", "  1match2", "  2match2", "  1match2", "nomatch2"] do
+      expect(subject).to be_a(Array)
+      insist { subject.size } == 2
+      insist { subject[0]["message"] } == "  match1\n  match1\nnomatch1"
+      insist { subject[1]["message"] } == "  1match2\n  2match2\n  1match2\nnomatch2"
+    end
+  end
+
+  describe "keep duplicates by default on message field" do
+    config <<-CONFIG
+      filter {
+        multiline {
+          pattern => "^\s"
+          what => "next"
+          add_tag => ["multi"]
+        }
+      }
+    CONFIG
+
+    sample ["  match1", "  match1", "nomatch1", "  1match2", "  2match2", "  1match2", "nomatch2"] do
+      expect(subject).to be_a(Array)
+      insist { subject.size } == 2
+      insist { subject[0]["message"] } == "  match1\n  match1\nnomatch1"
+      insist { subject[1]["message"] } == "  1match2\n  2match2\n  1match2\nnomatch2"
+    end
+  end
+
+  describe "remove duplicates using :allow_duplicates => false on message field" do
+    config <<-CONFIG
+      filter {
+        multiline {
+          allow_duplicates => false
+          pattern => "^\s"
+          what => "next"
+          add_tag => ["multi"]
+        }
+      }
+    CONFIG
+
+    sample ["  match1", "  match1", "nomatch1", "  1match2", "  2match2", "  1match2", "nomatch2"] do
+      expect(subject).to be_a(Array)
+      insist { subject.size } == 2
+      insist { subject[0]["message"] } == "  match1\nnomatch1"
+      insist { subject[1]["message"] } == "  1match2\n  2match2\nnomatch2"
+    end
+  end
+
+  describe "keep duplicates only on @source field" do
+    config <<-CONFIG
+      filter {
+        multiline {
+          source => "foo"
+          pattern => "^\s"
+          what => "next"
+          add_tag => ["multi"]
```

Comment:
this test doesn't check the existence of the multi tag, so we can remove it (same happens in other tests).